### PR TITLE
Transaction detailed report improvements

### PR DESCRIPTION
**Pull request by philipr-za**
### Closes Issues:
Closes https://github.com/big-neon/bigneon/issues/117
Closes https://github.com/big-neon/bigneon/issues/118

### Description:
Added the improvements to the Transaction Detailed Report requested in the issues linked above.
* When opening the report for an Event the starting date on the date filter is set to the sales_start_time so no transactions will be missed.
* The search function now ignores the Start date on the date time filter. The start date is ignored if there is anything in the search box at all.

### Environment Variables
 * No change

### API requirements
Requires https://github.com/big-neon/bn-api/pull/1230

### Release Video Link:
https://cl.ly/10fbbf69005a

### CODE REVIEW

File: src/components/pages/admin/events/dashboard/reports/Index.js [hunk 1]
```diff
@@ -31,8 +31,8 @@ class Report extends Component {
 		Bigneon()
 			.events.read({ id: eventId })
 			.then(response => {
-				const { name } = response.data;
-				this.setState({ eventName: name });
+				const { name, sales_start_date } = response.data;
```
Comment (Krakaw) [inline src/components/pages/admin/events/dashboard/reports/Index.js]: Just as an fyi you can redefine a variable name in one step like so:
```suggestion
				const { name, sales_start_date: salesStart } = response.data;
```

Comment (philipr-za) [reply src/components/pages/admin/events/dashboard/reports/Index.js]: Good to know, though in this case I am choosing a new name in in this.setState(...) in the next line anyway?

**Comment by Krakaw** [comment]
Just waiting for big-neon/bn-api#1230 to get merged in before merging this in.

File: src/components/pages/admin/events/dashboard/reports/Index.js (hunk 1 above)
Comment (Krakaw) [reply src/components/pages/admin/events/dashboard/reports/Index.js]: Yup, just some syntactic sugar for future use if you want.

Comment (philipr-za) [reply src/components/pages/admin/events/dashboard/reports/Index.js]: Thanks, there was another place where I needed that actually.

**Pull request merged**